credentials/credentials.go

```diff
@@ -193,6 +194,31 @@ func RequestInfoFromContext(ctx context.Context) (ri RequestInfo, ok bool) {
 	return
 }
 
+// AddressInfo contains address related data attached to the context passed to
+// ClientHandshake. This makes it possible for balancer implementations to pass
+// arbitrary data to the handshaker. Individual credential implementations
+// control the actual format of the data that they are willing to receive.
+//
+// This API is experimental.
+type AddressInfo struct {
+	// Attr is a generic key/value store.
+	Attr *attributes.Attributes
```

Comment:
resolver.Address instead?

Also, should we pick a more generic name for the struct?  This will allow us to pass any data we want to ClientHandshake.  (Boring, and also limited in what it can be used for, so maybe not the best option: "ClientHandshakeInfo"?)

Reply:
Package `resolver` depends on package `credentials`. So, this would case a cyclic dependency. If we can pull the `Address` struct out of the resolver package, we could do this.

Reply:
How about `HandshakerInfo` or `HandshakeInfo`. It's not much better than your suggestion :), but we can have a common struct as long as the client and server infos don't differ.

Reply:
I think they're different, actually.  Servers don't have `resolver.Addresses` (they don't have resolvers at all), so we probably don't want to share this.

Reply:
Done. Renamed to `ClientHandshakeInfo`.